staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go

```diff
@@ -84,12 +86,72 @@ type getterFunc func(ctx request.Context, name string, req *http.Request, trace
 // MaxRetryWhenPatchConflicts is the maximum number of conflicts retry during a patch operation before returning failure
 const MaxRetryWhenPatchConflicts = 5
 
+// Returns extracted opentracing.Span, or nil in case of any problems
+func startSpanFromRequestContext(req *http.Request, operationName string) opentracing.Span {
+	var serverSpan opentracing.Span
+	appSpecificOperationName := operationName
+	wireContext, err := opentracing.GlobalTracer().Extract(
+		opentracing.HTTPHeaders,
+		opentracing.HTTPHeadersCarrier(req.Header))
+	if err != nil {
+		if err != opentracing.ErrSpanContextNotFound {
+			glog.Warningf("Failed to read client context: %v", err)
+		}
+		return nil
+	}
+
+	serverSpan = opentracing.StartSpan(
+		appSpecificOperationName,
+		opentracing.ChildOf(wireContext),
+		opentracing.StartTime(time.Now()))
+
+	return serverSpan
+}
+
+func startTracing(req *http.Request, operationName string) (otSpan opentracing.Span, simpleTrace *utiltrace.Trace) {
+	trace := utiltrace.New(operationName)
```

Comment:
I wanted to get rid of the old trace library, and I figured that this was was easier.